module3/src/com/epam/regularexpressions/TextParser.java

```diff
@@ -0,0 +1,108 @@
+            for (String sentence : sentences) {
+                String[] words = splitWords(sentence);
+                for (int i = words.length - 1; i >= 0; i--) {
+                    for (int j = 0; j < i; j++) {
+                        if (words[j].length() > words[j + 1].length()) {
+                            String swap = words[j];
+                            words[j] = words[j + 1];
+                            words[j + 1] = swap;
+                        }
+                    }
+                }
+                for (String  word : words) {
+                    resultString.append(word);
+                    resultString.append(" ");
+                }
+                resultString.append("\b. ");
+            }
+            resultString.append("\n");
+        }
+        return resultString;
+    }
+
+    public static StringBuilder sortLexemes(String text, String lexeme) {
+        StringBuilder resultString = new StringBuilder();
+        String [] paragraphs = text.split("\n");
+        for (String paragraph : paragraphs) {
+            String[] sentences = splitSentences(paragraph);
+            for (String sentence : sentences) {
+                String[] words = splitWords(sentence);
+                for (int i = words.length - 1; i >= 0; i--) {
+                    for (int j = 0; j < i; j++) {
+                        int countLexemesFirstWord = 0;
+                        int countLexemesSecondWord = 0;
+                        for (int n = 0; n < words[j].length(); n++) {
+                            if (String.valueOf(words[j].charAt(n)).compareToIgnoreCase(lexeme) == 0) {
+                                countLexemesFirstWord++;
+                            }
+                        }
+                        for (int n = 0; n < words[j + 1].length(); n++) {
+                            if (String.valueOf(words[j + 1].charAt(n)).compareToIgnoreCase(lexeme) == 0) {
+                                countLexemesSecondWord++;
+                            }
+                        }
+                        if (countLexemesFirstWord < countLexemesSecondWord) {
+                            String tmp = words[j];
+                            words[j] = words[j + 1];
+                            words[j + 1] = tmp;
+                        } else if (countLexemesFirstWord == countLexemesSecondWord) {
+                            String [] forCompare = {words[j], words[j + 1]};
+                            Arrays.sort(forCompare);
+                            words[j] = forCompare[0];
+                            words[j + 1] = forCompare[1];
+                        }
+                    }
+                }
+                for (String  word : words) {
+                    resultString.append(word);
+                    resultString.append(" ");
+                }
+                resultString.append("\b. ");
+            }
+            resultString.append("\n");
+        }
+        return resultString;
+    }
```

Comment:
you need to try not to do such big methods. It must be split into separate ones. But okay, you will do this in the next module.